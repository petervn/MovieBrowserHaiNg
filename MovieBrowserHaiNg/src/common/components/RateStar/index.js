import React from 'react';
import {View, Image} from 'react-native';
import StartIcon from '../../asset/images/star.png';
import style from './style';

export default RateStar = ({voteAverage = 0}) => {
  if (isNaN(voteAverage)) {
    return null;
  }
  const average = Math.round(voteAverage);
  const length = Math.round(average * 0.5);

  return (
    <View style={style.container}>
      {[...Array(length)].map((e, i) => (
        <Image key={i} style={{height: 18, width: 18}} source={StartIcon} />
      ))}
    </View>
  );
};
