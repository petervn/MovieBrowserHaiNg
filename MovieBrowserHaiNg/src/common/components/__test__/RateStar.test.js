import React from 'react';
import {Image} from 'react-native';
import {shallow} from 'enzyme';
import {expect as chaiExpect} from 'chai';
import RateStar from '../RateStar';

describe('RateStar', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<RateStar voteAverage={8} />);
      expect(component).toMatchSnapshot();
    });
    it('should render correctly number of starts', () => {
      let component = shallow(<RateStar voteAverage={8} />);
      chaiExpect(component.find(Image)).to.have.lengthOf(4);
      component = shallow(<RateStar voteAverage={0} />);
      chaiExpect(component.find(Image)).to.have.lengthOf(0);
      component = shallow(<RateStar voteAverage={10} />);
      chaiExpect(component.find(Image)).to.have.lengthOf(5);
      component = shallow(<RateStar voteAverage={9} />);
      chaiExpect(component.find(Image)).to.have.lengthOf(5);
      component = shallow(<RateStar voteAverage={1} />);
      chaiExpect(component.find(Image)).to.have.lengthOf(1);
    });
  });
});
