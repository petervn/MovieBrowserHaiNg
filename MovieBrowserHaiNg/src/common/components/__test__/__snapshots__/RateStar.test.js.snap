// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RateStar Rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Image
    key="0"
    source={
      Object {
        "testUri": "../../../src/common/asset/images/star.png",
      }
    }
    style={
      Object {
        "height": 18,
        "width": 18,
      }
    }
  />
  <Image
    key="1"
    source={
      Object {
        "testUri": "../../../src/common/asset/images/star.png",
      }
    }
    style={
      Object {
        "height": 18,
        "width": 18,
      }
    }
  />
  <Image
    key="2"
    source={
      Object {
        "testUri": "../../../src/common/asset/images/star.png",
      }
    }
    style={
      Object {
        "height": 18,
        "width": 18,
      }
    }
  />
  <Image
    key="3"
    source={
      Object {
        "testUri": "../../../src/common/asset/images/star.png",
      }
    }
    style={
      Object {
        "height": 18,
        "width": 18,
      }
    }
  />
</View>
`;
